src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -633,8 +639,7 @@ static int parse_request(uint8_t code, uint8_t len, const void *option, void *us
                 break;
         case SD_DHCP_OPTION_MAXIMUM_MESSAGE_SIZE:
                 if (len == 2)
-                        req->max_optlen = be16toh(*(be16_t*)option) -
-                                          - sizeof(DHCPPacket);
+                        req->max_optlen = unaligned_read_be16(option) + sizeof(DHCPPacket);
```

Comment:
Let's wait for a comment from the systemd folk. If your presumption is correct, I will amend the last commit and force-push it.